### PR TITLE
Added Trufflehog to .travis.yml

### PR DESCRIPTION
**Pull request by GPelayo**
Note: The `trufflehog-rules.json` is a modified version of [the json file](https://github.com/HumanCellAtlas/dcplib/blob/master/components/trufflehog_regex_patterns.json). 

Also in `trufflehog-rules.json`, the private key rules (ex. RSA private key) have been edited to prevent from Trufflehog to identify these rules as secrets.

### CODE REVIEW

File: rules.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    "SSH (DSA) private key": "-----BEGIN DSA PRIVATE KEY-----[^[]",
+    "SSH (EC) private key": "-----BEGIN EC PRIVATE KEY-----[^[]",
+    "PGP private key block": "-----BEGIN PGP PRIVATE KEY BLOCK-----[^[]",
+    "Facebook Oauth": "[f|F][a|A][c|C][e|E][b|B][o|O][o|O][k|K].*['|\"][0-9a-f]{32}['|\"]",
+    "Twitter Oauth": "[t|T][w|W][i|I][t|T][t|T][e|E][r|R].*['|\"][0-9a-zA-Z]{35,44}['|\"]",
+    "GitHub": "[g|G][i|I][t|T][h|H][u|U][b|B].*['|\"][0-9a-zA-Z]{35,40}['|\"]",
+    "Google Oauth": "(\"client_secret\":\"[a-zA-Z0-9-_]{24}\")",
+    "AWS API Key": "AKIA(?!IOSFODNN7EXAMPLE)[0-9A-Z]{16}",
+    "Heroku API Key": "[h|H][e|E][r|R][o|O][k|K][u|U].*[0-9A-F]{8}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{4}-[0-9A-F]{12}",
+    "Generic Secret": "[s|S][e|E][c|C][r|R][e|E][t|T].*['|\"][0-9a-zA-Z]{32,45}['|\"]",
```
Comment (hannes-ucsc) [inline rules.json]: This regex makes no sense to me. Either I am the idiot or the person who wrote them.

`[s|S]` matches `s`, `S` or `|`. Is that really the intention?

Comment (GPelayo) [reply rules.json]: No, you're not the idiot. I think they meant (s|S).

Comment (hannes-ucsc) [reply rules.json]: Well, my confidence in this solution is pretty much gone. How can I trust a piece of security software if their authors can't even get a regular expression right?

**Comment by jessebrennan** [comment]
To be fair, I feel like most regular expressions written do not do what
they were intended to do.

File: .travis.yml [hunk 2]
```diff
@@ -18,12 +18,14 @@ install:
...
   - pip install -r requirements.dev.txt
   - pip install coveralls codecov
+  - pip install trufflehog
 
 script:
   # Travis VM images contain a /etc/boto.cfg which ultimately causes moto to look for the Python bindings for GCE.
   # Setting BOTO_CONFIG seems to override that.
   - export BOTO_CONFIG=/dev/null
   - make test
+  - trufflehog --regex --rules trufflehog-rules.json --entropy=False file:///home/travis/build/DataBiosphere/azul
```
[the remark this answers is not in the document]
Comment (hannes-ucsc) [reply .travis.yml]: Remind me to talk to you about `hash trufflehog`.

**Pull request merged**